cmd/recover.go

```diff
@@ -0,0 +1,123 @@
+package cmd
+
+import (
+	"bytes"
+	"math/rand"
+	"strconv"
+	"sync"
+	"syscall"
+
+	"github.com/juicedata/juicefs/pkg/meta"
+	"github.com/juicedata/juicefs/pkg/utils"
+	"github.com/urfave/cli/v2"
+)
+
+func cmdRecover() *cli.Command {
+	return &cli.Command{
+		Name:      "recover",
+		Action:    recover,
+		Category:  "ADMIN",
+		Usage:     "recover files from trash",
+		ArgsUsage: "META HOUR ...",
+		Description: `
+Rebuild the tree structure for trash files, and move them back to original directories.
+
+Examples:
+$ juicefs recover redis://localhost/1 2023-10-10-01`,
+		Flags: []cli.Flag{
+			&cli.BoolFlag{
+				Name:  "move-to-original",
+				Usage: "move the recovered files into original directory",
+			},
+			&cli.IntFlag{
+				Name:  "threads",
+				Value: 10,
+				Usage: "number of threads",
+			},
+		},
+	}
+}
+
+func recover(ctx *cli.Context) error {
```

Comment:
this function name conflicts with build-in function `recover() error`.

Reply:
and the debug command try to use the build-in `recover` https://github.com/juicedata/juicefs/blob/main/cmd/debug.go#L242